com.unity.multiplayer.mlapi/Runtime/Messaging/InternalMessageHandler.cs

```diff
@@ -60,7 +60,15 @@ internal static void HandleConnectionRequest(ulong clientId, Stream stream)
 
                 if (NetworkManager.Singleton.NetworkConfig.ConnectionApproval)
                 {
-                    byte[] connectionBuffer = reader.ReadByteArray();
+                    byte[] connectionBuffer = null;
+                    if (stream.Position == stream.Length)
+                    {
+                        NetworkLog.LogWarning($"Connection approval could be aborted for client ID {clientId} due to missing authorization token data.  Make sure the client is setting the NetworkConfig.ConnectionData before sending the authorization response.");
```

Comment:
Just to confirm.
- NetworkConfig by default is the *empty* array of size 0 and will work just fine. There is no need for the user to manually assign ConnectionData in code. This exception does not happen to me if I just use connection approval without setting a ConnectionData by code.
![image](https://user-images.githubusercontent.com/43687322/110114101-0106c580-7dac-11eb-83e8-61df31001170.png)


There are two cases where this warning will be thrown.
1. If a user manually overrides NetworkConfig.ConnectionData to null this here will be triggered. I don't see why anyone would ever do that though but it's good to capture this case.
2. If a client connects with ConnectionApproval disabled (because they are using a wrong build or different scene with different config) the stream will also be at it's end and the warning will be thrown. We should include this in the warning message. I think this is the more common case.